File: guides/micronaut-oauth2-linkedin/java/src/main/java/example/micronaut/LinkedInApiClient.java
package example.micronaut;

import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Header;
import io.micronaut.http.client.annotation.Client;
import org.reactivestreams.Publisher;

@Client(id = "linkedin") // <1>
public interface LinkedInApiClient {

    @Get("/v2/me") // <2>
    Publisher<LinkedInMe> me(@Header String authorization); // <3> <4>
}

File: guides/micronaut-oauth2-linkedin/python/src/example/micronaut/linkedin_api_client.py
from abc import ABC, abstractmethod
from typing import Annotated

from micronaut.core.async_.annotation import SingleResult
from micronaut.http import HttpHeaders
from micronaut.http.annotation import Get, Header
from micronaut.http.client.annotation import Client
from org.reactivestreams import Publisher

from .linkedin_me import LinkedInMe


@Client(id="linkedin")  # <1>
class LinkedInApiClient(ABC):

    @Get("/v2/me")  # <2>
    @SingleResult
    @abstractmethod
    def me(
        self,
        authorization: Annotated[str, Header(HttpHeaders.AUTHORIZATION)],  # <4>
    ) -> Publisher:  # <3>
        ...
